### PR TITLE
Fix VS restore on sdk.sln and clean up nuget packaging of dependencies

### PR DESCRIPTION
**Pull request by nguerrera**
Cherry picking some incremental improvements from a big build refactoring in progress that will get us to dogfooding test explorer, dotnet test, and dotnet pack. The larger change still needs a lot of work.

Issues addressed in this round

1 Restore not working in VS: 
 * VS restore was not wired to the same package folder as command line.
 * VS did not like how we specified TargetFrameworks in some cases

2. Packaging of dependencies
 * We let the build step copy the right dlls (with some trickery to get the filtering right) and then package what comes out instead of maintaining a separate list of  nuget package content to replicate in our package.

@natidea @dsplaisted

### CODE REVIEW

File: build/Nuget/Microsoft.NET.Sdk.Nuget.targets [hunk 1]
```diff
@@ -3,155 +3,16 @@
...
-      <CopyLocalBuildFile Include="@(CopyLocalBuildCrossTargetingFile)" />
-      <CopyLocalBuildFile Include="@(CopyLocalSdkFile)" />
-      <CopyLocalBuildFile Include="@(CopyLocalSatelliteAssembly)" />
-
-      <!-- Use the contents of the output directory of the net46 build of the SDK tasks, except
-           for MSBuild and Roslyn DLLs and their dependencies-->
-      <Net46OutputAssembly Include="$(OutDir)net46\*.dll" />
-      
-      <!-- Use the versions of these that come with MSBuild -->
-      <Net46OutputAssembly Remove="$(OutDir)net46\Microsoft.Build.Framework.dll" />
```
Comment (natidea) [inline build/Nuget/Microsoft.NET.Sdk.Nuget.targets]: What is the new equivalent of these Removes?

Comment (nguerrera) [inline build/Nuget/Microsoft.NET.Sdk.Nuget.targets]: For net46, ExcludeAssets=Runtime. For netcoreapp, that + FilterCopyLocal.

Basically we adjust the build step to deploy only the right assemblies and then packaging can copy them wholesale.

**Review by natidea — APPROVED** [review]

**Review by dsplaisted — APPROVED** [review]
Did you do a diff of the package layout before and after this change?

**Review by livarcocc — APPROVED** [review]

**Comment by nguerrera** [comment]
> Did you do a diff of the package layout before and after this change?

The only differences are:

1. These facades are no longer copied to the net46 folder, but those seem left over from when we deployed netstandard1 libraries there. I looked through the closure in ILSpy of the whole net46 folder in it's all plain desktop assemblies referencing mscorlib and friends now. This is good cleanup.
```
  file source="$(OutputPath)PackagesLayout\tools\net46\System.Console.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.Diagnostics.Process.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.IO.FileSystem.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.IO.FileSystem.Primitives.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.Runtime.InteropServices.RuntimeInformation.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.Security.Claims.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.Security.Principal.Windows.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.Threading.Thread.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.Xml.ReaderWriter.dll"
  file source="$(OutputPath)PackagesLayout\tools\net46\System.Xml.XmlDocument.dll"
```
2. We got moved from the netstandard1.3 to the netstandard1.6 version of Microsoft.Extensions.DependencyModel in tools\netcoreapp1.0 folder. This is also fine.

**Comment by srivatsn** [comment]
Didn't we explicitly add System.IO.FileSystem because MSBuild doesn't have the facades in it's bin

**Comment by nguerrera** [comment]
At that time, we were deploying a netstandard1.3 version of our dll. Now we multi-target so we don't depend on the facade nor does anything in our closure.

**Pull request merged**